src/Page/Submission/Team2/Analysis.js

```diff
@@ -0,0 +1,260 @@
+// Anthony Ilersich
+// Evelyn Law
+// Lilian Chen
+// Aaron Kong
+
+//read in stock json
+var data = require("./../../../Data/stocks.json");
+
+//constructor for Analysis class, 
+function Analysis() {
+	
+	//used to record statistics about each stock
+	this.profiles = [];
+	
+	//call to begin analysis of stocks
+	this.run = function() {
+		var start = new Date();
+		this.profiles = generateProfiles();
+		console.log("generated " + Object.keys(data).length + " profiles in " + (new Date() - start) + "ms!");
+		console.log(this.profiles);
+	}
+}
+
+//export Analysis to make it available to other files
+export default Analysis;
+
+//calculates the simple moving average over a dataset, over the last n samples
+function SMA(dataset, n) {
+	var result = [];
+	for (var i = 0; i < dataset.length; i++) {
+		var sum = 0, m = 0;
+		for (var j = Math.max(i - n + 1, 0); j <= i; j++) {
+			sum += dataset[j];
+			m++;
+		}
+		result.push(sum / m);
+	}
+	return result;
+}
+
+//calculates the exponential moving average over a dataset, over the last n
+//samples
+function EMA(dataset, n) {
+	var result = [];
+	var avg = undefined;
+	for (var i = 0; i < dataset.length; i++) {
+		if (avg === undefined) {
+			avg = dataset[i];
+		}
+		else {
+			avg = (avg * (n - 1) + dataset[i]) / n;
+		}
+		result.push(avg);
+	}
+	return result;
+}
+
+//generates an array of values of a particular property of a stock over the
+//dataset
+function getValues(stock, propKey) {
+	var values = [];
+	for (var i = 0; i < stock.length; i++) {
+
+		values.push(stock[i][propKey]);
+	}
+	return values;
+}
+
+//calculates average daily return over the whole dataset for a given stock,
+//using the closing price
+function getAvgAnnualReturn(stock) {
+	
+	//assume there are 365 days in a year
+	const daysPerYear = 365;
+	
+	//sum daily returns
+	var sum = 0;
+	for (var i = 1; i < stock.length; i++) {
+		
+		//read current and previous closing price
+		var c0 = stock[i - 1].Close;
+		var c1 = stock[i].Close;
+		
+		//skip this term if we have bad values
+		if (c0 === null || c1 === null) {
+			continue;
+		}
+		
+		//add daily return;
+		sum += (stock[i].Close - stock[i - 1].Close) / stock[i - 1].Close;
+	}
+	
+	//take average
+	var avg = sum / (stock.length - 1);
+	
+	//calculate yearly return, return value
+	return avg * daysPerYear;
+}
+
+//calculates compounding daily return for a given stock
+function getCompoundReturn(stock) {
+	
+	//get necessary values
+	var beginningBalance = stock[0].Close; 
```

Comment:
Use adj close instead of Close price since close price would not adjust the event of stock split or dividends paying.